src/manifold/src/boolean3.cpp

```diff
@@ -578,6 +578,10 @@ Boolean3::Boolean3(const Manifold::Impl &inP, const Manifold::Impl &inQ,
       Intersect12(inQ, inP, s20, p2q0, s11, p1q1, z20, xyzz11, p2q1_, false);
   PRINT("x21 size = " << x21_.size());
 
+  if (x12_.size() + x21_.size() >= std::numeric_limits<int>::max() ||
+      w03_.size() + w30_.size() >= std::numeric_limits<int>::max())
```

Comment:
Thinking about this, we can probably remove w03 and w30 too: they always just have size of P.NumVert() and Q.NumVert() respectively.

Reply:
ah, make sense